.github/workflows/ci.yml

```diff
@@ -6,6 +6,23 @@ on:
   pull_request:
 
 jobs:
+  type_check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought:** I wonder do we want to run this on all supported python versions as well? 

Typing should be a non-runtime behaviour, so I don't think it will matter, and the imports should be caught by the test execution anyway so I don't think we would need it? Just thinking out loud

Reply:
I guess you are right. I have removed the Python version requirement as checking with MyPy should not really be tied to a specific Python version.